src/gnssro/gnssro_bufr2ioda2.f90

```diff
@@ -0,0 +1,620 @@
+integer   :: nlev_dimid
+integer   :: varid_geo_temp,varid_geo_pres,varid_geo_shum,varid_geo_geop, varid_geo_geop_sfc
+integer   :: grpid_metadata,grpid_obserror,grpid_obsvalue
+character(len=256)        :: infile, outfile
+character,dimension(8)    :: subset
+character(len=10)         :: anatime
+integer(i_kind)           :: ndatetime = 20
+character(len=20)         :: datetime
+integer(i_kind)           :: i,k,m,ireadmg,ireadsb,said,siid,ptid,sclf,asce,ogce
+integer(i_kind)           :: lnbufr    = 10
+integer(i_kind)           :: nread,ndata,nvars,nrec, ndata0
+integer(i_kind)           :: anatime_i
+integer(i_kind)           :: idate5(6), idate,iadate5(6)
+integer(i_kind)           :: mincy,minobs
+integer(i_64)             :: epochtime
+
+logical                   :: good,outside
+integer                   :: refflag, bendflag
+integer(i_kind),parameter :: mxib=31
+integer(i_kind)           :: ibit(mxib),nib
+integer(i_kind),parameter :: maxlevs=500
+integer(i_kind),parameter :: n1ahdr=13
+integer(i_kind)           :: maxobs
+real(r_kind) :: timeo
+type gpsro_type
+      integer(i_kind), allocatable, dimension(:)    :: said
+      integer(i_kind), allocatable, dimension(:)    :: siid
+      integer(i_kind), allocatable, dimension(:)    :: sclf
+      integer(i_kind), allocatable, dimension(:)    :: ptid
+      integer(i_kind), allocatable, dimension(:)    :: recn
+      integer(i_kind), allocatable, dimension(:)    :: asce
+      integer(i_kind), allocatable, dimension(:)    :: ogce
+      real(r_kind), allocatable, dimension(:)       :: time
+      character(len=20), allocatable, dimension(:)  :: datetime
+      integer(i_64),allocatable, dimension(:)     :: epochtime
+      real(r_kind), allocatable, dimension(:)     :: lat
+      real(r_kind), allocatable, dimension(:)     :: lon
+      real(r_kind), allocatable, dimension(:)     :: rfict
+      real(r_kind), allocatable, dimension(:)     :: azim
+      real(r_kind), allocatable, dimension(:)     :: geoid
+      real(r_kind), allocatable, dimension(:)     :: msl_alt
+      real(r_kind), allocatable, dimension(:)     :: ref
+      real(r_kind), allocatable, dimension(:)     :: refoe_gsi
+      real(r_kind), allocatable, dimension(:)     :: bend_ang
+      real(r_kind), allocatable, dimension(:)     :: impact_para
+      real(r_kind), allocatable, dimension(:)     :: bndoe_gsi
+end type gpsro_type
+
+type(gpsro_type) :: gpsro_data
+
+real(r_kind),dimension(n1ahdr)     :: bfr1ahdr
+real(r_kind),dimension(50,maxlevs) :: data1b
+real(r_kind),dimension(50,maxlevs) :: data2a
+real(r_kind),dimension(maxlevs)    :: nreps_this_ROSEQ2
+integer(i_kind)                    :: iret,levs,levsr,nreps_ROSEQ1,nreps_ROSEQ2_int
+real(r_kind) :: pcc,qfro,usage,dlat,dlat_earth,dlon,dlon_earth,freq_chk,freq,azim
+real(r_kind) :: height,rlat,rlon,ref,bend,impact,roc,geoid,  bend_error,ref_error,bend_pccf,ref_pccf
+real(r_kind) :: obsErr
+real(r_kind),   parameter :: missingvalue=-9.9e10
```

Comment:
@hailingz @rhoneyager Is there any default netcdf or IODA missing values to use here instead?

Reply:
@srherbener

Reply:
You can use any fill value that you want, but you need to inform the ioda reader what that value is. This is done by adding the attribute "_FillValue" to each integer and floating point variable where the value of the attribute is the fill value you decided to use.

Reply:
yes, I now use the same number as you have in the sample data, which is the biggest number of the data type.